index.html

```diff
@@ -1044,48 +1060,101 @@
             <li>Return <var>P</var>, but continue running these steps in
             parallel.
             </li>
-            <li>For each <var>known connection</var> in the <a>set of
-            presentations</a>,
-              <div style="margin-left: 2em">
-                If <a>presentation connection state</a> of <var>known
-                presentation</var> is not <a for=
-                "PresentationConnectionState">terminated</a>, the
-                <a>presentation URL</a> of <var>known connection</var> is equal
-                to the <a>presentation request URL</a> of
-                <var>presentationRequest</var>, and the <a>presentation
-                identifier</a> of <var>known connection</var> is equal to
-                <var>presentationId</var>, run the following steps:
-                <ol>
-                  <li>Let <var>S</var> be the <a>presentation connection</a> of
-                  <var>known connection</var>.
-                  </li>
-                  <li>
-                    <a>Resolve</a> <var>P</var> with <var>S</var>.
-                  </li>
-                  <li>
-                    <a>Queue a task</a> to <a>fire</a> a <a>trusted event</a>
-                    with the name <a>connectionavailable</a>, that uses the
-                    <a>PresentationConnectionAvailableEvent</a> interface, with
-                    the <a for=
-                    "PresentationConnectionAvailableEvent">connection</a>
-                    attribute initialized to <var>S</var>, at
-                    <var>presentationRequest</var>. The event must not bubble,
-                    must not be cancelable, and has no default action.
-                  </li>
-                  <li>
-                    <a>Establish a presentation connection</a> with
-                    <var>S</var>.
-                  </li>
-                  <li>Abort all remaining steps.
-                  </li>
-                </ol>
-              </div>
+            <li>Search the <a>set of controlling presentations</a> for a
+            <a>PresentationConnection</a> that meets the following criteria:
+            its <a>controlling browsing context</a> is the current <a>browsing
+            context</a>, its <a>presentation connection state</a> is not
+              <a for="PresentationConnectionState">terminated</a>, its
+              <a>presentation URL</a> is equal to the <a>presentation request
+              URL</a> of <var>presentationRequest</var>, and its
+              <a>presentation identifier</a> is equal to
+              <var>presentationId</var>.
+            </li>
+            <li>If such a <a>PresentationConnection</a> exists, run the
```

Comment:
Perhaps you mean PresentationConnection object is shared among multiple contexts?  However, we would have to think carefully about when it is safe to delete a connection.  Sharing would also make the close algorithm more complicated.